### PR TITLE
Release pull request for branch 20.4

### PR DESCRIPTION
**Pull request by robot-clickhouse**
This PullRequest is part of ClickHouse release cycle. It's used by CI system only. Don't perform any changes with it.

### CODE REVIEW

**Comment by nikitamikhaylov** [comment]
Stress tests (memory) fixed in master, usban assert in lz4 fixed in master

**Comment by KochetovNicolai** [comment]
Only `test_default_role` integration tests failed.

File: src/Storages/MergeTree/MergeTreeDataMergerMutator.cpp [hunk 1]
```diff
@@ -794,6 +794,12 @@ MergeTreeData::MutableDataPartPtr MergeTreeDataMergerMutator::mergePartsToTempor
     if (need_remove_expired_values)
         merged_stream = std::make_shared<TTLBlockInputStream>(merged_stream, data, new_data_part, time_of_merge, force_ttl);
 
+    if (data.hasSkipIndices())
+    {
+        merged_stream = std::make_shared<ExpressionBlockInputStream>(merged_stream, data.skip_indices_expr);
+        merged_stream = std::make_shared<MaterializingBlockInputStream>(merged_stream);
+    }
```
Comment (azat) [inline src/Storages/MergeTree/MergeTreeDataMergerMutator.cpp:801]: But #11162 has been already reverted from 20.4 in d2ee7dc0f2ec617ff6dbec7658964d3e6e5368f3, this means that there is no test for this change right now, can you pick the test too then?

**Comment by alesapin** [comment]
All important tests should be fixed, but flaky failures still may happen.

**Comment by alesapin** [comment]
Memory stress:
```
2020.07.14 08:33:10.377725 [ 51650 ] {} <Fatal> BaseDaemon: ########################################
2020.07.14 08:33:10.377864 [ 51650 ] {} <Fatal> BaseDaemon: (version 20.4.7.67 (official build)) (from thread 4789) (query_id: 16f55d4c-af63-4a84-8be1-6179741904b5) Received signal Aborted (6).
2020.07.14 08:33:10.377943 [ 51650 ] {} <Fatal> BaseDaemon: 
2020.07.14 08:33:10.378020 [ 51650 ] {} <Fatal> BaseDaemon: Stack trace: 0x7fd879c473eb 0x7fd879c26899 0x9827797 0x98261c1 0x97c9f3f
2020.07.14 08:33:10.378188 [ 51650 ] {} <Fatal> BaseDaemon: 4. gsignal @ 0x463eb in /usr/lib/x86_64-linux-gnu/libc-2.30.so
2020.07.14 08:33:10.378277 [ 51650 ] {} <Fatal> BaseDaemon: 5. abort @ 0x25899 in /usr/lib/x86_64-linux-gnu/libc-2.30.so
2020.07.14 08:33:10.382322 [ 51650 ] {} <Fatal> BaseDaemon: 6. ? @ 0x9827797 in /usr/bin/clickhouse
2020.07.14 08:33:10.386267 [ 51650 ] {} <Fatal> BaseDaemon: 7. ? @ 0x98261c1 in /usr/bin/clickhouse
2020.07.14 08:33:10.390218 [ 51650 ] {} <Fatal> BaseDaemon: 8. ? @ 0x97c9f3f in /usr/bin/clickhouse
```
Seems like we don't have some changes for memory sanitizer?

**Comment by alesapin** [comment]
ubsan 
```../contrib/lz4/lib/lz4.c:818:44: runtime error: applying zero offset to null pointer```
Missing a patch in lz4?

**Comment by alesapin** [comment]
DatabaseAtomic failures not important, experimental functionality.

**Comment by alesapin** [comment]
Functional tests with memory sanitizer -- `01081_demangle`. Strange error, but it works in the release build, so skipsit.

**Comment by alesapin** [comment]
Stress tests with address sanitizer:
```
==273==ERROR: AddressSanitizer: allocator is out of memory trying to allocate 0x20000 bytes
==273==ERROR: AddressSanitizer failed to allocate 0x2000 (8192) bytes of InternalMmapVector (error code: 12)
```
It's ok, we have fixed this error in master, but I don't remember where and when.

**Comment by alesapin** [comment]
ClickHouse Unit Tests Msan -- infinite recursion in libunwind.

**Comment by alesapin** [comment]
No integration tests. Run them manually.

**Comment by alesapin** [comment]
Asan has known failure
```
==273==ERROR: AddressSanitizer: allocator is out of memory trying to allocate 0x20000 bytes
==273==ERROR: AddressSanitizer failed to allocate 0x2000 (8192) bytes of InternalMmapVector (error code: 12)
ERROR: Failed to mmap```

**Comment by alesapin** [comment]
`test_moves_to_disk_work` no fix in test in this branch. It's ok.

**Pull request closed without merging**